sunpy/map/sources/sdo.py

```diff
@@ -48,11 +48,15 @@ def __init__(self, data, header, **kwargs):
 
         # Fill in some missing info
         self.meta['detector'] = self.meta.get('detector', "AIA")
+        if 'pixlunit' in self.meta:
```

Comment:
```suggestion
        if 'bunit' not in self.meta and 'pixlunit' in self.meta:
```
This guards against a user set 'bunit' being overwritten.